p2p/src/sync/mod.rs

```diff
@@ -118,42 +110,34 @@ where
         log::info!("Starting SyncManager");
 
         let mut new_tip_receiver = self.subscribe_to_new_tip().await?;
-        self.is_initial_block_download =
-            self.chainstate_handle.call(|c| c.is_initial_block_download()).await??;
+        self.is_initial_block_download.store(
+            self.chainstate_handle.call(|c| c.is_initial_block_download()).await??,
+            Ordering::SeqCst,
+        );
 
         loop {
             tokio::select! {
-                event = self.messaging_handle.poll_next() => match event? {
-                    SyncingEvent::Connected { peer_id } => {
-                        self.register_peer(peer_id).await?;
-                    },
-                    SyncingEvent::Disconnected { peer_id } => {
-                        self.unregister_peer(peer_id);
-                    },
-                    SyncingEvent::Message { peer, message } => {
-                        let res = self.handle_message(peer, message).await;
-                        self.handle_result(peer, res).await?;
-                    },
-                    SyncingEvent::Announcement{ peer, announcement } => {
-                        let res = self.handle_announcement(peer, *announcement).await;
-                        self.handle_result(peer, res).await?;
-                    }
-                },
                 block_id = new_tip_receiver.recv() => {
                     // This error can only occur when chainstate drops an events subscriber.
                     let block_id = block_id.ok_or(P2pError::ChannelClosed)?;
                     self.handle_new_tip(block_id).await?;
                 },
-                (peer, block) = async { self.blocks_queue.pop_front().expect("The block queue is empty") }, if !self.blocks_queue.is_empty() => {
-                    let res = self.send_block(peer, block).await;
-                    self.handle_result(peer, res).await?;
-                }
+
+                // Resend messages from peers to the backend.
+                message = self.peer_receiver.recv() => {
+                    let (peer, message) = message.ok_or(P2pError::ChannelClosed)?;
+                    self.messaging_handle.send_message(peer, message)?;
+                },
+
+                event = self.messaging_handle.poll_next() => {
+                    self.handle_peer_event(event?)?;
+                },
```

Comment:
`handle_peer_event` will return an error if some peer's channel is disconnected.
So at this line we will stop the sync manager's even loop if some peer's loop is stopped (for any reason).
Looks like it's OK right for now. But maybe it would be better to ignore the error here if the processing for some peer has been stopped unexpectedly?

Reply:
I would prefer this to be handled in this PR correctly.

Reply:
Sure, fixed.